sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/WithFailures.java

```diff
@@ -92,6 +93,24 @@
     }
   }
 
+  /**
+   * A handler that holds onto the {@link Throwable} that led to the exception, returning it along
+   * with the original value as a {@link KV}.
+   *
+   * <p>Extends {@link SimpleFunction} so that full type information is captured. {@link KV} and
+   * {@link ComparableThrowable} coders can be easily inferred by Beam, so coder inference can be
+   * successfully applied if the consuming transform passes type information to the failure
+   * collection's {@link TupleTag}. This may require creating an instance of an anonymous inherited
+   * class rather than of this class directly.
+   */
+  public static class ThrowableHandler<T>
```

Comment:
Yeah, this is just for dealing with boilerplate. More specifically, since we offer a way to get the exception's values as a map, it seemed good to provide a handler that preserves the original. Otherwise, it seems like we're heavily encouraging using the map when it may not always be the best option.